### ISSUE TITLE
Cura 4.4.1 - Profile and Printer settings/values sometimes (not always) revert back

### ISSUE THREAD

**Issue by rplant45a** [Ultimaker/Cura#7111]
**Application version** - 4.4.1
(The version of the application this issue occurs with.)

**Platform** - Win 10 
see image attached

![about my laptop](https://user-images.githubusercontent.com/49108341/74608091-327ef700-50ac-11ea-91b2-63f0b8040a35.jpg)
(Information about the operating system the issue occurs on. Include at least the operating system and maybe GPU.)

**Printer** - Creality CR-10
(Which printer was selected in Cura?)

**Reproduction steps**
nothing specific, I changed 
1. from skirt to brim and when I loaded Cura again it had reverted back to skirt.
2. start gcode by changing some for EZABL code to raise hot end higher before it went to home and printed the two lines to clear the nozzle. Also changed the z offest.  Both of these gcode changes revered back to previous instances.

**Screenshot(s)** - I don't think it's needed but if you want a shot of my gcode I'll be happy to supply but it's just the EZABL with two insignificant changes
(Image showing the problem, perhaps before/after images.) 

**Actual results** - like I said cura reverted back to previous instances of the about 2 changes
(What happens after the above steps have been followed.)

**Expected results** - should not have reverted
(What should happen after the above steps have been followed.)

**Project file**
(For slicing bugs, provide a project which clearly shows the bug, by going to File->Save. For big files you may need to use WeTransfer or similar file sharing sites.)

**Log file**
(See https://github.com/Ultimaker/Cura#logging-issues to find the log file to upload, or copy a relevant snippet from it.)

**Additional information**
(Extra information relevant to the issue.)

**Comment by Ghostkeeper** [comment]
I'm suspecting that maybe Cura didn't close properly and wasn't able to save its settings? You could try this by waiting for 10 seconds after you made the setting change, as it should autosave every 10 seconds. You can also verify that this happened by looking in the log for the entry "Autosaving preferences, instances and profiles".

I haven't been able to reproduce this. I don't think it's specific to a certain printer or profile, as our code tends to be generic for that sort of thing.

**Comment by Vandrasc** [comment]
I am also not able to reproduce this issue. @rplant45a are you also experiencing this in the latest Cura 4.5?

**Comment by rplant45a** [comment]
No has did not happen again after I created the post.  But like I said it happened a couple time too many.  But, I did upgrade to 4.5.1 and have not had any issues at all.  Could be that I indeed crashed the instance and it reverted back to pre profile… being a sw person myself, I could see how this could happen and be a fail safe… and if the case, it would be good to have msg about trouble reading current profile (file read error) and reverting to possible know good profile… but thanks for checking!!!!!!!!!!!  You guys are doing such a great job and I know it’s probably on your own time!!!!!!!!  So thanks a bunch!!!!!!!!!

 

 

Thanks…  Rick

 

Richard Plant

919-349-7890

104 Parktree Court

Cary, NC 27519

ישו הוא מלך
